provider/waiter.go

```diff
@@ -0,0 +1,160 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-cty/cty"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+)
+
+// Waiter is a simple interface to implement a blocking wait operation
+type Waiter interface {
```

Comment:
The thinking here is to keep all of the waiting logic away from the plan/apply operations so that it just boils down to a function call that composes something with a `Wait()` function on it. This is likely going to grow in scope to include rollout, waiting for conditions, job completion, ingress load balancers etc. We should be able to keep all this when migrate to using the SDK.